sql/hive/src/main/scala/org/apache/spark/sql/hive/TableReader.scala

```diff
@@ -159,36 +159,11 @@ class HadoopTableReader(
     def verifyPartitionPath(
         partitionToDeserializer: Map[HivePartition, Class[_ <: Deserializer]]):
         Map[HivePartition, Class[_ <: Deserializer]] = {
-      if (!sparkSession.sessionState.conf.verifyPartitionPath) {
```

Comment:
after this pr https://github.com/apache/spark/pull/17187， read hive table which does not use `stored by` will not use `HiveTableScanExec`.

this function has  a bug ,that if the partition path is custom path
1. it will still do filter for all partition path in the parameter `partitionToDeserializer`,
2. it will scan the path which does not belong to the table ,e.g. custom path is `/root/a`
and the partitionSpec is `b=1/c=2`, this will lead to scan `/` because of the `getPathPatternByPath `